apps/Launcher/qml/main.qml

```diff
@@ -33,16 +34,40 @@ Rectangle {
             id: menu
             width: Style.menuWidth * root.width
             height: root.height
-            demoItemVisible: demoServiceUrl && demoServiceImageFolder && demoServiceDeflectHost
-            onClearSession: sendJsonRpc("controller", "clear");
-            onStartWebbrowser: sendJsonRpc("application", "browse", "");
-            onPoweroffScreens: sendJsonRpc("application", "poweroff");
-            onStartWhiteboard: sendJsonRpc("application", "whiteboard");
-            onShowFilesPanel: centralWidget.sourceComponent = fileBrowser
-            onShowSessionsPanel: centralWidget.sourceComponent = sessionsBrowser
-            onShowSaveSessionPanel: centralWidget.sourceComponent = saveSessionPanel
-            onShowDemosPanel: centralWidget.sourceComponent = demoLauncher
-            onShowOptionsPanel: centralWidget.sourceComponent = optionsPanel
+            demoItemVisible: demoServiceUrl && demoServiceDeflectHost
+            onClearSession: sendJsonRpc("controller", "clear")
+            onStartWebbrowser: sendJsonRpc("application", "browse", "")
+            onPoweroffScreens: sendJsonRpc("application", "poweroff")
+            onStartWhiteboard: sendJsonRpc("application", "whiteboard")
+            onShowFilesPanel: {
+                if (demoLauncherWidget.visible)
+                    demoLauncherWidget.visible = false
```

Comment:
no need to check, just do:
demoLauncherWidget.visible = false

In fact, it would even be better to avoid duplication by doing something like:
`centralWidget.onSourceComponentChanged(): demoLauncherWidget.visible = false`